src/app/route/MyAccountOrders/MyAccountOrders.component.js

```diff
@@ -0,0 +1,166 @@
+
+class MyAccountOrders extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            finishedLoading: false
+        };
+    }
+
+    componentDidMount() {
+        this.requestCustomerOrders();
+        this.updateBreadcrumbs();
+    }
+
+    requestCustomerOrders() {
+        const { requestCustomerOrders } = this.props;
+
+        return requestCustomerOrders().then(() => this.setState({ finishedLoading: true }));
+    }
+
+    updateBreadcrumbs() {
+        const { updateBreadcrumbs } = this.props;
+        const breadcrumbs = [
+            {
+                url: '/my-account-orders',
+                name: 'My Account Orders'
+            },
+            {
+                url: '/',
+                name: 'Home'
+            }
+        ];
+
+        updateBreadcrumbs(breadcrumbs);
+    }
+
+    renderOrderRows() {
+        const { orders, currency } = this.props;
+        const { finishedLoading } = this.state;
+
+        if (finishedLoading && currency) {
+            return orders.map(order => (
+                <tr key={ order.id }>
+                    <td>{ order.id }</td>
+                    <td>{ order.created_at }</td>
+                    <td>
+                        { order.grand_total }
+                        { currency }
+                    </td>
+                    <td>{ order.status }</td>
+                    <td>View Order</td>
+                </tr>
+            ));
+        }
+
+        return (
+            <tr>
+                <td>{ <TextPlaceholder length="medium" /> }</td>
+                <td>{ <TextPlaceholder length="medium" /> }</td>
+                <td>{ <TextPlaceholder length="medium" /> }</td>
+                <td>{ <TextPlaceholder length="medium" /> }</td>
+                <td><TextPlaceholder length="medium" /></td>
+            </tr>
+        );
+    }
+
+    renderOrderInformation() {
+        const { orders } = this.props;
+        const { finishedLoading } = this.state;
+        const showHeader = !finishedLoading || (finishedLoading && !!orders.length);
+
+        return (
+            <>
+                <h1 block="MyAccountOrders" elem="Heading">My Orders</h1>
+                <table block="MyAccountOrders" elem="Table" mods={ { isVisible: showHeader } }>
+                    <thead>
+                        <tr>
+                            <th>Order #</th>
+                            <th>Date</th>
+                            <th>Order Total</th>
+                            <th>Status</th>
+                            <th>Action</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        { this.renderOrderRows() }
+                    </tbody>
+                </table>
+                { finishedLoading && !orders.length
+                    && <div block="MyAccountOrders" elem="Empty">You currently have no orders!</div>
```

Comment:
Make it multiline please, move into separate function. Check `finishedLoading && !orders.length` in that function.